src/main/java/edu/pdx/cs/sftp/User.java

```diff
@@ -44,93 +51,97 @@
+   * @return <code>true</code> if the string is a valid password <code>false</code> otherwise.
    */
-  private boolean verifyPassword(String password) {
-    return !password.isEmpty() && !password.contains(" ");
+  private boolean verifyPassword(String passwordToVerify) {
+    return !passwordToVerify.isEmpty() && !passwordToVerify.contains(" ");
   }
 
   /**
-   * Prompts the user for a valid username (usernames must be alphanumeric of size 8-20)
+   * Prompt the user for a valid username.
+   * Username must be alphanumeric of size 8-20.
    *
-   * @return username input
+   * @return username input.
    */
   String getUsername() {
-    System.out.println("Enter your username:");
+    out.println("Enter your username:");
     username = scanner.next();
-    // username = ""; //for testing
-    if (username == null || username.isEmpty() || !verifyUsername(username)) {
-      System.err.println(
-          "That was not a valid username.  Please enter 8-20 alphanumeric " + "characters.");
-      username = scanner.next(); // comment out for testing
+    while (username == null || username.isEmpty() || !verifyUsername(username)) {
+      err.println( "The username is invalid.\n"
+                 + "Please enter 8-20 alphanumeric characters.");
+      username = scanner.next();
     }
     return username;
   }
 
   /**
-   * Checks if the string is a valid username
+   * Verify the username is 8-20 alphanumeric characters.
    *
-   * @param toVerify a string to be checked against a regular expression
-   * @return <code>true</code> if the string is a valid username <code>false</code> otherwise
+   * @param usernameToVerify is the string to be checked against a regular expression.
+   * @return <code>true</code> if the string is a valid username <code>false</code> otherwise.
    */
-  boolean verifyUsername(String toVerify) {
-    String userNamePattern = "^(?=.{2,20}$)(?![_.])(?!.*[_.]{2})[a-zA-Z0-9._]+(?<![_.])$";
-    Pattern pattern = Pattern.compile(userNamePattern);
-    Matcher matcher = pattern.matcher(toVerify);
+  boolean verifyUsername(String usernameToVerify) {
+    String usernamePattern = "^(?=.{2,20}$)(?![_.])(?!.*[_.]{2})[a-zA-Z0-9._]+(?<![_.])$";
+    Pattern pattern = Pattern.compile(usernamePattern);
+    Matcher matcher = pattern.matcher(usernameToVerify);
     return matcher.matches();
   }
 
   /**
-   * Prompts the user for a valid host name which follows the following parameters: (1) Must be
-   * alphanumeric + "." (2) Can't be longer than 255 characters (3) Host name segments cannot exceed 63
-   * characters (4) Must end and start with alphanumeric characters
+   * Prompt the user for a valid host name, which follows the below parameters:
+   * (1) Must be alphanumeric
+   * (2) Can't be longer than 255 characters
+   * (3) Host name segments cannot exceed 63 characters
+   * (4) Must end and start with alphanumeric characters
    *
-   * @return The host name input
+   * @return host name input.
```

Comment:
```suggestion
   * @return hostname input.
```
I legitimately Googled this (ha!), but I think it should be `hostname` instead of `host name` (see: https://english.stackexchange.com/questions/100858/hostname-or-host-name).